### PR TITLE
Omit unnecessary string registration for HillFort and fix popup window crash for miniHillFort

### PR DESCRIPTION
**Pull request by MichalZr6**

### CODE REVIEW

**Comment by MichalZr6** [comment]
No, although I tried to do some investigation on that one. Will see more into the issue a bit later today.

File: lib/mapObjectConstructors/HillFortInstanceConstructor.cpp [hunk 1]
```diff
@@ -18,7 +18,9 @@ VCMI_LIB_NAMESPACE_BEGIN
 void HillFortInstanceConstructor::initTypeData(const JsonNode & config)
 {
 	parameters = config;
-	VLC->generaltexth->registerString(parameters.getModScope(), TextIdentifier(getBaseTextID(), "unavailableUpgradeMessage"), parameters["unavailableUpgradeMessage"].String());
+	if(getModScope() != "core")
```
Comment (IvanSavenko) [inline lib/mapObjectConstructors/HillFortInstanceConstructor.cpp:21]:
```suggestion
	if(!parameters["unavailableUpgradeMessage"].isNull())
```

Comment (MichalZr6) [reply lib/mapObjectConstructors/HillFortInstanceConstructor.cpp]: Done :)

**Comment by MichalZr6** [comment]
This PR resolves #4904.

**Pull request merged**